src/pyconnect/pyconnectsink.py

```diff
@@ -377,8 +378,20 @@ def _commit(self) -> None:
         if not offsets:
             logger.info("No offsets to commit.")
         else:
-            logger.info(f"Committing offsets: {offsets}")
-            self._consumer.commit(offsets=offsets, asynchronous=False)
+            max_attempts: int = 2
+            attempt_count: int = 1
+            while attempt_count <= max_attempts:
+                try:
+                    logger.info(f"Committing offsets: {offsets}")
+                    self._consumer.commit(offsets=offsets, asynchronous=False)
+                except KafkaException as ke:
+                    logger.error(
+                        f"Kafka exception occurred while comitting offsets (attempt {attempt_count}): {str(ke)}"
+                    )
+                    if attempt_count == max_attempts:
+                        exit(1)
```

Comment:
```suggestion
                        raise
```
PyConnect has handlers for cleaning up after itself, we should give it a chance to do so.